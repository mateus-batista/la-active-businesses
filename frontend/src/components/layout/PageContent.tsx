/** @jsx jsx */
import { css, jsx } from "@emotion/react";
import { ReactNode } from "react";
import { Title } from "./Title";

export interface PageContentProps {
  title: string;
  children: ReactNode;
}

export function PageContent(props: PageContentProps) {
  const { title, children } = props;
  return (
    <div>
      <div
        css={css`
          height: 75px;
          display: flex;
          align-items: center;
          background-color: white;
        `}
      >
        <Title
          css={css`
            font-size: 1.5rem;
            margin: 0;
            width: 100vw;
          `}
        >
          {title}
        </Title>
      </div>
      <div
        css={css`
          margin-top: 2rem;
        `}
      >
        {children}
      </div>
    </div>
  );
}
