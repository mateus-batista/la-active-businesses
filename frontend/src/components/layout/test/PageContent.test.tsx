import { render, screen } from "@testing-library/react";
import React from "react";
import { PageContent } from "../PageContent";

describe("PageContent test", () => {
  test("should render correctly", () => {
    expect(
      render(
        <PageContent title="Test">
          <div>Test</div>
        </PageContent>
      )
    ).toMatchSnapshot();
  });

  test("should display given title and child", () => {
    render(
      <PageContent title="Test">
        <div>Child</div>
      </PageContent>
    );

    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(screen.getByText("Child")).toBeInTheDocument();
  });
});
