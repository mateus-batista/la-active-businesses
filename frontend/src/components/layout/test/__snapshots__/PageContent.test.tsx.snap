// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
}

.emotion-1 {
  margin: 0.5rem;
  color: #0f2940;
  text-align: center;
  font-size: 1.5rem;
  margin: 0;
  width: 100vw;
}

.emotion-2 {
  margin-top: 2rem;
}

<body>
    <div>
      <div>
        <div
          class="emotion-0"
        >
          <div
            class="emotion-1"
          >
            Test
          </div>
        </div>
        <div
          class="emotion-2"
        >
          <div>
            Test
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
}

.emotion-1 {
  margin: 0.5rem;
  color: #0f2940;
  text-align: center;
  font-size: 1.5rem;
  margin: 0;
  width: 100vw;
}

.emotion-2 {
  margin-top: 2rem;
}

<div>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          Test
        </div>
      </div>
      <div
        class="emotion-2"
      >
        <div>
          Test
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
